### PR TITLE
markdown: fix redundant leading spaces in markdown list

### LINKED ISSUE #6948: Redundant leading spaces are not removed from multi-line items in markdown lists
**Prettier 1.19.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAVAFnATnABAQx1xgHcJcBnAB3zDgtwDMIAbFiEgSygHNc5a6XAAN8AOgkATYbgBGcdiSQAdKKoC0BXLlXa523XrAGoeySe1xVGvfgvbZ94ztPbzLvf2tQAjGK0eDvbazoa47oZWar7+AWGOgaGu4cFeUCAANCAQVDCc0BTIoIRYHAAKhAiFKPgAbhCckpkgsli0ANZwMADKNGDcPMgwWACucFnoMAC2LADq6Jzw1LRw3VWLnLWLAJ7I4BSFWdwU2DBlbTxT+MiM+CwnWQBWFAAeAEJtYJ09+FNwADLcOA3O4PEDPF7dAYsOAARRGEHgIPu4xANCwJyweyuWHakg46SyVCw3Bgs0aMHQyAAHAAGImlE6zNpUPbE+jYWrArIARwR8HOOWqIHwFHUUDgcEkUuaOD5nBw53wl2uSFuKKyJymnCGo1RFGhcP5wLVoNRMHwsnJkkpyAATFlhvhOCwBgBhCBTK57KDQbkgEYnVCW6rqsG1MYASSg0tg3TAJNyAEEY90YNsYciTgBfbNAA)
```sh
--parser markdown
```

**Input:**
```markdown
There are two spaces following each `a...d` below:

- a  
   b  
    c  
   d  
  e

-   a  
   b  
  c  
   d  
    e

1. a  
   b  
    c  
   d  
  e

1.  a  
   b  
  c  
   d  
    e

```

**Output:**
```markdown
There are two spaces following each `a...d` below:

- a  
   b  
   c  
   d  
  e

- a  
   b  
  c  
   d  
   e

1. a  
    b  
    c  
    d  
   e

1. a  
    b  
   c  
    d  
    e

```

**Expected behavior:**

```markdown
There are two spaces following each `a...d` below:

- a  
  b  
  c  
  d  
  e

- a  
  b  
  c  
  d  
  e

1. a  
   b  
   c  
   d  
   e

1. a  
   b  
   c  
   d  
   e
```

The bug seems to occur only when we have `[space][space]` or `\n` before end of line. More examples:
* [unordered lists](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBiVACAchDMBOAhgJYA2xUA5hgM4AOhYcNGFtEArvkxgGYT4MkKADc4+GBWoADADqyo0vLnqNmrKBjr44I4nADuChQFoMGaHAw6AJhyg3CsDKTiEbU2gx7QMhUqQuFMwK5n6h5gBGEUIxNjFwxlBm5jAGuLb2js6u7p6qTCy+-oHkUCGaYYQxGNGV5mA18fUYiVCmGCKE+ACenrkeVF5qNBHV9XVhsfXN5m0d5l29-W6D1AXMADR4hADWVmkQNeNhk9Nhs2HzUOgYACrpeERk+d7q-ILlBkHlLGzaun0RnayXMlmscDsDicMBcq1eagsmhKP3UGBihHMMUiWPqYFxFwJcySKTwj0y0Jy8KGGyKyICqJY6JamOZUxxbLC+M55hsROJILMSz6QwGCMKY35HJq3LiROupOFKzyNLeNG2MD2B3SxylRNlM35rSSClud2eZRk8iUHwwX0ZGi0Oj0hhJYPKEKh2VhYtViOKDLKIXk9WqIdO8hijXDvMj9QVYUOnqyMLhKvWaqRfkDwVGMfMYZq0XzsRL8RLCaVoupGZGYzjEZL0biDeNgrCVeovtrhQ1WrJRxZrdqrebM2HbZAmxAEDokmgNGQoG6+AgBgACt0EIuUIQRBBiDYpyBIkQwPsYABlbxSZAEDhwacACxgAFtSAB1J-EeAbS-bn9iD0GAemQcAaEXacKBocQYHXIhKFfQhkF4fwYOnAArGgAA8ACEzwvS9CFfOAABlghQtDHxALDsMvKRXAARQ4CB4Eo0h0JABh8Bg-AwKQ-BdhsNcoGPbQKBgD9DxgJ9kAADgABmnbQIBgj8iDoMCAV4sRjwARxY+B4NnHcQEIGgTHKSFIWPHQDOIHR4MIRDkKQVCOOomDX2IO98AfacaAYuBmNYuB2M4zVIikmwZOQAAmacCBIS0AGEIFfJCwKgSxjw4GC7kISId3cziRAfABJBwECvMB8GIOcAEEHEvEDXHCuAAF8OqAA)
* [ordered lists](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBiVACAchDMBOAhgJYA2xUA5hgM4AOhYcNGFtEArvkxgGYT4MkKADc4+GBWoADADqyo0vLnqNmrKBjr44I4nADuChQEYAdBmhwMOgCYcotwrAyk4hW1NoMe0DIVJSVwpmBQxwwjDwgCMooTjbOLhjKHNwmANcOwcnFzcPL1UmFj8AoPIoUM1w-ziMWOrwsDrExoxkqFMLEUJ8AE8vfM8qbzUaKMjGhpr4xtbwjq7wnv7B92HqIuYAGjxCAGtrDIg6yZrp2Zr5msWodAwAFUy8IjJCn3V+QUqDYMqWNjaXT6IydVIWKw2OD2RzOGCudbvNSWTRlP7qDBxQjhOLRHGNMD4q5EhYpNJ4Z7ZWF5REjLYlVGBdEsTFtbGsmZ4jk1Qnc8K2EmksFpFYDEZDJHFCaCrl1XkJEm3cmitYFOkfGi7GAHI6ZU4yknyuaC9opBT3B6vCoyeRKL4YH7MjRaHR6QxkiGVKEw3LwiXq5GlJkVULyRqRMPneRxZqR-nRxpKiwUrLQnJwhFqzYalH+YMhcZxiIJzklnll+NF03C7q9MXUf3ZsYTCuxKuxhJlpPLOuqjajYpanUp06tssduYV6sgbYgCB0STQGjIUC9fAQAwABV6CGXKEIIggxFsM5A0SIYEOMAAyj4pMgCBw4LOABYwAC2pAA6i-iPAtteu5-sQegwH0yDgDQy6zhQNDiDAm5EJQ76EMgvABHBs4AFY0AAHgAQheV7XoQ75wAAMiEaEYc+IA4bh15SG4ACKHAQPA1GkJhIAMPgcH4BBKH4PstgblAp7aBQMBfseMAvsgAAcAAMs7aBAcFfkQdAQUC-FiKeACObHwIh857iAhA0AAtJU0LQqeOhGcQOiIYQyGoUg6FcbRcHvsQD74E+s40ExcCsexcCcdx2rRDJthycgABMs4ECQ1oAMIQO+KEQVAVinhwcEPIQ0R7l53EiE+ACSjgIDeYD4MQC4AIKONeYFuFFcAAL7dUAA)
* [real-world case](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBiVACAygVwA54QBO8AJhggG4CWR0AtgjBpQIZHWsBGANnAM4AdKMIC0GAAYAhAIIBhANIBRAHIARAPoAJAPJYAKhIwZhxjABkIYVjGrQMEAGYYYACzgYurMAGsE5fjgiSiCXCAw8OgAPAE8MIjgARxwBGH4wgDpTDDlWKE8PHChSOEdqKDhyR2IMfh9qAnKAc2ExSSkAJR0AdSwlDqMTfIwpOgB3QKIwjFYcGAh6G2prHh44iDwEDHKMEsoMjABVQMkoaDgjeYc8GFEIOdaocQkABR0Ow2NsszMX4mZ7GNXMtXNMeLMoGBQW4PHsZsUInQAkEQlMeFZWKsYsIQAAaEAbWzQfjIUDscYvdgIEkoViUCDUUh4kBcIjePwwLB4bzNZAwIgpfGuGD0HjdYHwfjcsBwLDU6i2GgwGLIcD8En48qTGAvNlNRbIRyYwL4gBW-Cio3ZcE5rEY5nKcENxrgZotWGafAAijgIPBnTwTSBuURJqrFkQfKQIGMoMzIuUYN1GW5kAAOAAM+MiEEC3TZeFVkQEKKd+OSfrguo2NJArH4ogqlUqzISyVoVb1BqQRsDrpAgXo1D5Av7-E9cB9lYDQZg3GTpFTSAATPj+axqDxmnIFt2QGcKsycIF9Nwab2g5QUgBJYpMLBgDg3GTFLDKvgzuAAXy-QA)

### PR DESCRIPTION
**Pull request by sasurau4**
fix #6948 

- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by kachkaev** [comment]
Changes are user-facing, so it’d be great to see `changelog_unreleased/*/pr-XXXX.md` ;)

**Comment by sasurau4** [comment]
Thanks for reviewing!
I added the changelog_unreleased file!

File: changelog_unreleased/markdown/pr-7178.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+#### markdown: fix rebundant leading spaces in markdown list ([#7178](https://github.com/prettier/prettier/pull/7178) by [@sasurau4](https://github.com/sasurau4))
```
Comment (sosukesuzuki) [inline changelog_unreleased/markdown/pr-7178.md]: CI was failed for typo. Please fix it.
```suggestion
#### markdown: fix redundant leading spaces in markdown list ([#7178](https://github.com/prettier/prettier/pull/7178) by [@sasurau4](https://github.com/sasurau4))
```

Comment (sasurau4) [reply changelog_unreleased/markdown/pr-7178.md]: THX!

File: changelog_unreleased/markdown/pr-7178.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
+#### markdown: fix redundant leading spaces in markdown list ([#7178](https://github.com/prettier/prettier/pull/7178) by [@sasurau4](https://github.com/sasurau4))
```
Comment (sosukesuzuki) [inline changelog_unreleased/markdown/pr-7178.md]: The prefix for the language is unnecessary in changelog. Please fix it in your another PRs too.
```suggestion
#### Fix redundant leading spaces in markdown list ([#7178](https://github.com/prettier/prettier/pull/7178) by [@sasurau4](https://github.com/sasurau4))
```

**Pull request merged**